bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -44,45 +47,86 @@
 import io.github.hapjava.accessories.HomekitAccessory;
 import io.github.hapjava.server.impl.HomekitRoot;
 import io.github.hapjava.server.impl.HomekitServer;
+import io.github.hapjava.server.impl.crypto.HAPSetupCodeUtils;
 
 /**
  * Provides access to openHAB items via the HomeKit API
  *
  * @author Andy Lintner - Initial contribution
  */
-@Component(service = { Homekit.class }, configurationPid = "org.openhab.homekit", property = {
+@Component(service = { Homekit.class }, configurationPid = HomekitSettings.CONFIG_PID, property = {
         Constants.SERVICE_PID + "=org.openhab.homekit", "port:Integer=9123" })
 @ConfigurableService(category = "io", label = "HomeKit Integration", description_uri = "io:homekit")
 @NonNullByDefault
 public class HomekitImpl implements Homekit {
     private final Logger logger = LoggerFactory.getLogger(HomekitImpl.class);
-    private final NetworkAddressService networkAddressService;
-    private final HomekitChangeListener changeListener;
 
-    private HomekitSettings settings;
+    private final @NonNullByDefault({}) NetworkAddressService networkAddressService;
+    private final @NonNullByDefault({}) ConfigurationAdmin configAdmin;
+
+    private @NonNullByDefault({}) HomekitAuthInfoImpl authInfo;
+    private @NonNullByDefault({}) HomekitSettings settings;
```

Comment:
```suggestion
    private HomekitAuthInfoImpl authInfo;
    private HomekitSettings settings;
```

Reply:
thx, good one. 
i have not thought about @NonNullByDefault({}) after changed the injection to constructor injection